ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobStoreHardDelete.java

```diff
@@ -82,17 +82,20 @@ public MessageInfo getMessageInfo(Read read, long offset, StoreKeyFactory storeK
             return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
                 Utils.addSecondsToEpochTime(properties.getCreationTimeInMs(), properties.getTimeToLiveInSeconds()));
           } else {
-            boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(stream);
-            return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
-                deleteFlag);
+            DeleteRecord deleteRecord = MessageFormatRecord.deserializeDeleteRecord(stream);
+            if (deleteRecord.getVersion() == MessageFormatRecord.Delete_Version_V1) {
+              return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(), true);
```

Comment:
I am not sure if size would be deterministic in future. Lets say we come up with V3 which has same size as V2. So, I don't think size should determine the version.